### PR TITLE
Recursively search for stack config under stacks directory

### PR DESCRIPTION
**Pull request by jimcroft**
Convenience feature to allow stack files to exist anywhere under the stacks directory.

Much like template files can exists anywhere under the templates directory our users would like the same behavior in stacks. It allows them to organize stack files however they see fit.

There might be a downside or problem doing this but not sure what that'd be. :)

### CODE REVIEW

File: lib/sfn-parameters/stacks.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@ class ParametersStacks < ParametersInfrastructure
       # @param stack_name [String]
       # @return [Smash]
       def load_file_for(stack_name)
-        root_path = config.fetch(:sfn_parameters, :directory, 'stacks')
+        root_path = config.fetch(:sfn_parameters, :directory, 'stacks', '**')
```
Comment (chrisroberts) [inline lib/sfn-parameters/stacks.rb]: This modification would change the configuration location for the directory path from:

```ruby
{sfn_parameters: {directory: 'PATH'}}
```

with `"stacks"` being the default directory to:

```ruby
{sfn_parameters: {directory: {stacks: 'PATH'}}}
```

with `"*"` being the default value. The `Smash#fetch` method works much like the `Hash#dig` method (just before `#dig` was available). Were you thinking of something like:

```ruby
root_path = config.fetch(:sfn_parameters, :directory, "stacks#{File::DELIMITER}*")
```

**Review by chrisroberts — CHANGES_REQUESTED** [review]

**Comment by chrisroberts** [comment]
@jimcroft Hi! So the problem this would introduce is that the file names correspond to specific stacks. This is how sfn knows which file to load when creating/updating a stack. Can you describe the type of behavior you're trying to accomplish? I understand the ability to let the files be located anywhere, but if the file name does not match the stack name I'm not sure how we can determine which file to apply.

**Comment by jimcroft** [comment]
Thanks for the feedback Chris.

So the stack filename itself would still have to match the stack name, no change there. It's more that we want to have an arbitrary directory structure under stacks. So something like this would be possible:

```
/stacks
  /dev
    dev-web.rb
    dev-db.rb
  /prod
    prod-web.rb
    prod-db.rb
``` 

This is very much a cosmetic-would-be-nice feature (so no big deal if kicked to the kerb) but does cause a bit of confusion when devs can create templates, components, etc. where ever they like (so long as the template/component name is unique).

**Comment by siulcode** [comment]
Hey guys, 

I know this is an old thread, but I am kinda hitting something similar. 

Is there any way to inject custom configuration files (yaml) in app directories into sparkle? 

Below is my directory structure, we were using a similar directory structure with a different tool, but a new requirement changed with the need of the "Cloud-agnostic" feature, hence we ended up using SparkleFormation and it seems to be an awesome tool so far. 
 
Structure: 
`├───components
├───conf
│   ├───apps
│   │   ├───app1 >> [EC2,SG,LB].yaml
│   │   └───app2 >> [EC2,SG,LB].yaml
│   └───environment >> main.yaml
├───dynamics
└───registry
`
If there's anything else we can do or any other way to achieve this it would be greatly appreciated!

Thanks again for the great support.

cc.: @chrisroberts

**Pull request closed without merging**